qiskit_nature/second_q/mappers/tapered_qubit_mapper.py

```diff
@@ -0,0 +1,177 @@
+        return returned_op
+
+    def map_clifford(
+        self,
+        second_q_ops: SparseLabelOp | ListOrDictType[SparseLabelOp],
+    ) -> PauliSumOp | ListOrDictType[PauliSumOp]:
+        """Maps a second quantized operator or a list, dict of second quantized operators based on
+        the internal mapper. Then, composes all mapped pauli operators with the clifford operations
+        defined by the internal ``Z2Symmetries`` to prepare for the symmetry reduction.
+        This composition gives isospectral operators and exposes redundant qubits for later tapering.
+
+        Args:
+            second_q_ops: A second quantized operator, or list (resp. dict) thereof.
+
+        Returns:
+            A qubit operator in the form of a PauliSumOp, or list (resp. dict) thereof if a list
+            (resp. dict) of second quantized operators was supplied.
+        """
+        wrapped_type = type(second_q_ops)
+
+        if issubclass(wrapped_type, SparseLabelOp):
+            second_q_ops = [second_q_ops]
+
+        wrapped_second_q_ops: _ListOrDict[SparseLabelOp] = _ListOrDict(second_q_ops)
+
+        qubit_ops: _ListOrDict = _ListOrDict()
+        for name, second_q_op in iter(wrapped_second_q_ops):
+            qubit_ops[name] = PauliSumOp(self._map_clifford_single(second_q_op))
+
+        returned_ops: PauliSumOp | ListOrDictType[PauliSumOp]
+        returned_ops = qubit_ops.unwrap(wrapped_type)
+
+        return returned_ops
+
+    def taper_clifford(
+        self,
+        pauli_ops: PauliSumOp | ListOrDictType[PauliSumOp],
+        *,
+        check_commutes: bool = True,
+        suppress_none: bool = False,
+    ) -> PauliSumOp | ListOrDictType[PauliSumOp]:
+        """Applies the symmetry reduction on a ``PauliSumOp`` or a list (resp. dict). This method implies
+        that the second quantized operators were already mapped to Pauli operators and composed with the
+        clifford operations defined in the symmetry, for example using the ``map_clifford`` method.
+
+        Args:
+            pauli_ops: A pauli operator already evolved with the symmetry clifford operations.
+            check_commutes: If the commutativity of operators with symmetries must be checked before
+                any calculation.
+            suppress_none: If None should be placed in the output list where an operator
+                did not commute with symmetry, to maintain order, or whether that should
+                be suppressed where the output list length may then be smaller than the input.
+
+        Returns:
+            A qubit operator in the form of a PauliSumOp, or list (resp. dict) thereof if a list
+            (resp. dict) of second quantized operators was supplied.
+        """
+        wrapped_type = type(pauli_ops)
+
+        if issubclass(wrapped_type, PauliSumOp):
+            pauli_ops = [pauli_ops]
+            suppress_none = False
+
+        wrapped_pauli_ops: _ListOrDict[PauliSumOp] = _ListOrDict(pauli_ops)
```

Comment:
I wonder whether we can also de-duplicate this boilerplate code via a `_ListOrDict.wrap` method which returns both `wrapped_type` and `wrapped_ops`.

The question would be how to deal with `suppress_none` but I think this should be manageable, too.

Reply:
I'm thinking of a wrap(xx, suppress_none) which would write onto a private attribute suppress_none of the class _ListOrDict. 
Additionally, I think the suppress_none for single operators is actually not needed. We can see in the unwrap() method that True/False would give the same result when wrapped_type is not List/Dict. I feel like it is similar to a comment above, i.e. something that was kept but the underlying logic has changed. I'll check that

Reply:
Similarly, we may want wrapped_type to be an attribute of _ListOrDict. I'm not sure though that we want to complexify this class.